### ISSUE TITLE
[Feature Request] Accessibility Improvement / Tooltip Support

### ISSUE THREAD

**Issue by veeru153** [lucide-icons/lucide#477]
I shifted to Lucide from Feather after having no way to add tooltips directly. A similar issue (https://github.com/feathericons/react-feather/issues/67) exists for Feather which references this project but I didn't see any request or any implementation here yet. The gist of this issue/request is the ability to add alternative text like this:

`<Icon alt="Navigate to Report" />`

This would allow browsers to display a tooltip when you hover the cursor over the icon like this:

![image](https://user-images.githubusercontent.com/32760644/150055714-9a4fff67-824d-48de-9441-b64a8072fd12.png)

The linked issue goes into more detail. The icons work really well as buttons and this addition could improve the accessibility of the websites using these icons directly.

**Comment by ericfennis** [comment]
@veeru153 Thanks for opening.
Is `<Icon  title="Delete" />` not working mentioned in that issue?

**Comment by veeru153** [comment]
Thank you for the quick response. 

No this does not work. After some devtooling, the reason seems to be that when we add a `title` prop to the icon, it gets passed as an attribute to the SVG element like this:

`<svg ... title="Delete">...</svg>`

In order to render the tooltip, the text should be a Title element nested inside the SVG element like this:

```
<svg ...>
  <title>Delete</title>
  ...
</svg>
```

**Comment by ericfennis** [comment]
@veeru153 Allrighty I get it. Also read some more about it. It also good for accessibility.

My proposal to accomplish this, is that we make it possible to pass children in icons. That creates also new possibilities to render something "over" the icon. This also keeps it really flexible and easy to implement.

## Example title and description
```svg
<Icon>
  <title>My awesome text</title>
  <desc>More detailed description about the icon</desc>
</Icon>
```

## Example rendering something over the icon, for example amount of items.
```svg
<ShoppingCart>
  <text>5</text>
</ShoppingCart>
```

**Comment by veeru153** [comment]
Ooh yes! While this is different from the traditional image `alt` attribute, this seems to be a better approach for the reasons you mentioned. I am assuming this would allow nesting icons inside icons too? Something like:

```
<Book>
  <Code />
</Book>
```

**Comment by ericfennis** [comment]
@veeru153 Yess correct!

**Comment by ericfennis** [comment]
Wanted to release this feature, but with a final test it doesn't work as expected. 
I tested it wrong, I tested it by using the browser inspector and changed the position of the title element below the children in the DOM. That worked for some reason, the tooltip showed when hovering. A bit stupid but lesson learned, always test on initial render. So @veeru153 you were right, `<title>` and `<desc>` need to be placed before the children to make it work.

Here a test environment:
https://codesandbox.io/s/tooltip-does-not-work-5jjh2e?file=/App.svelte

TLDR:
Current implementation doesn't work, we need to have a new solution.

**Comment by TheodorRene** [comment]
So there is currently no way to make the icons accessible to screen readers?

**Comment by tommy4st** [comment]
> So there is currently no way to make the icons accessible to screen readers?

It's actually possible by using the aria attributes. For example, you can use aria-label instead of alt or for decorative icons you can use aria-hidden. But I would still prefer to set the alt attribute.

**Comment by karsa-mistmere** [comment]
Resolved by #2122 & (partly) #2158
